flocker/control/_model.py

```diff
@@ -128,24 +150,17 @@ def __init__(self):
         """
         if self.maximum_retry_count is not None:
             if not isinstance(self.maximum_retry_count, int):
-                raise TypeError(
-                    "maximum_retry_count must be an integer or None, "
-                    "got %r" % (self.maximum_retry_count,))
+                return (False,
+                        "maximum_retry_count must be an integer or None, "
+                        "got %r" % (self.maximum_retry_count,))
```

Comment:
Hm I guess this does type enforcement.  Any reason not to switch this to built-in pyrsistent type enforcement?